src/presentation_exchange.rs

```diff
@@ -1,80 +1,584 @@
+    pub fn new(path: JsonPath) -> ConstraintsField {
         ConstraintsField {
-            path,
-            id,
-            purpose,
-            name,
-            filter,
-            optional,
-            intent_to_retain,
+            path: NonEmptyVec::new(path),
+            id: None,
+            purpose: None,
+            name: None,
+            filter: None,
+            optional: None,
+            intent_to_retain: None,
         }
     }
+
+    /// Add a new path to the constraints field.
+    pub fn add_path(mut self, path: JsonPath) -> Self {
+        self.path.push(path);
+        self
+    }
+
+    /// Return the paths of the constraints field.
+    ///
+    /// `path` is a non empty list of [JsonPath](https://goessner.net/articles/JsonPath/) expressions.
+    ///
+    /// For syntax definition, see: [https://identity.foundation/presentation-exchange/spec/v2.0.0/#jsonpath-syntax-definition](https://identity.foundation/presentation-exchange/spec/v2.0.0/#jsonpath-syntax-definition)
+    pub fn path(&self) -> &NonEmptyVec<JsonPath> {
+        &self.path
+    }
+
+    /// Set the id of the constraints field.
+    ///
+    /// The fields object MAY contain an id property. If present, its value MUST be a string that
+    /// is unique from every other field object’s id property, including those contained in other
+    /// Input Descriptor Objects.
+    pub fn set_id(mut self, id: String) -> Self {
+        self.id = Some(id);
+        self
+    }
+
+    /// Return the id of the constraints field.
+    pub fn id(&self) -> Option<&String> {
+        self.id.as_ref()
+    }
+
+    /// Set the purpose of the constraints field.
+    ///
+    /// If present, its value MUST be a string that describes the purpose for which the field is being requested.
+    pub fn set_purpose(mut self, purpose: String) -> Self {
+        self.purpose = Some(purpose);
+        self
+    }
+
+    /// Return the purpose of the constraints field.
+    pub fn purpose(&self) -> Option<&String> {
+        self.purpose.as_ref()
+    }
+
+    /// Set the name of the constraints field.
+    ///
+    /// If present, its value MUST be a string, and SHOULD be a human-friendly
+    /// name that describes what the target field represents.
+    ///
+    /// For example, the name of the constraint could be "over_18" if the field is a date of birth.
+    pub fn set_name(mut self, name: String) -> Self {
+        self.name = Some(name);
+        self
+    }
+
+    /// Return the name of the constraints field.
+    pub fn name(&self) -> Option<&String> {
+        self.name.as_ref()
+    }
+
+    /// Set the filter of the constraints field.
+    ///
+    /// If present its value MUST be a JSON Schema descriptor used to filter against
+    /// the values returned from evaluation of the JSONPath string expressions in the path array.
+    pub fn set_filter(mut self, filter: SchemaValidator) -> Self {
```

Comment:
I might be wrong, but it appears that `SchemaValidator` is not publically accessible outside of this crate.

Reply:
Made the module public.